met-web/src/components/engagement/view/PreviewBanner.tsx

```diff
@@ -10,50 +10,43 @@ export const PreviewBanner = () => {
     const navigate = useNavigate();
     const { engagementLoading, savedEngagement, publishEngagement } = useContext(ActionContext);
     const isLoggedIn = useAppSelector((state) => state.user.authentication.authenticated);
+    const [isDraft, setIsDraft] = useState(true);
 
     const engagementId = savedEngagement.id || '';
-    if (engagementLoading) {
-        return <Skeleton variant="rectangular" width="100%" height="10em" />;
-    }
-    const isDraft = savedEngagement.status_id === EngagementStatus.Draft;
 
-    const validate = () => {
-        const errors = {
-            status: savedEngagement.status_id !== EngagementStatus.Draft,
-        };
-
-        return Object.values(errors).some((isError: unknown) => isError);
-    };
+    useEffect(() => {
+        setIsDraft(savedEngagement.status_id === EngagementStatus.Draft);
+    }, [savedEngagement]);
 
     const handlePublishEngagement = async () => {
-        const hasErrors = validate();
-
-        if (!hasErrors) {
-            const engagement = await publishEngagement({
+        if (isDraft) {
+            await publishEngagement({
                 ...savedEngagement,
                 status_id: EngagementStatus.Published,
                 published_date: new Date().toUTCString(),
                 surveys: [],
             });
-
+        } else {
             navigate(`/`);
-
-            return engagement;
         }
     };
 
+    if (engagementLoading) {
+        return <Skeleton variant="rectangular" width="100%" height="10em" />;
+    }
+
     return (
-        <ConditionalComponent condition={isDraft && isLoggedIn}>
```

Comment:
if the whole component is conditional, might we consider returning early. if(isLoggedIn) return null above the long return one. This way the main logic of the component wouldn't have to be nested under this conditional component, reader would focus on its main logic.

Reply:
Updated